hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CustomizedScanInfoBuilder.java

```diff
@@ -34,8 +36,18 @@
 
   private KeepDeletedCells keepDeletedCells = null;
 
+  private Integer minVersions;
+
+  private final Scan scan;
+
   public CustomizedScanInfoBuilder(ScanInfo scanInfo) {
     this.scanInfo = scanInfo;
+    this.scan = new Scan();
+  }
+  public CustomizedScanInfoBuilder(ScanInfo scanInfo, Scan scan) throws IOException {
```

Comment:
The Scan constructor throws IOE now?? Not an issue for this patch, though

Reply:
I was surprised too.

Reply:
Looks like the copy constructor for Scan has thrown IOException back since 2009, but the current code does not actually contain anything that throws it. Since I assume changing the signature is more trouble than it's worth, I just had to rethrow from the CustomizedScanInfoBuilder constructor and getScan() methods.

Reply:
Let's just catch the IOException in the code here and then throw an AssertionError to indicate this should not happen, and we can filed another issue to remove the IOException from throws.